impeller/renderer/backend/vulkan/render_pass_vk.h

```diff
@@ -25,20 +25,92 @@ class RenderPassVK final : public RenderPass {
  private:
   friend class CommandBufferVK;
 
-  std::weak_ptr<CommandBufferVK> command_buffer_;
+  std::shared_ptr<CommandBufferVK> command_buffer_;
```

Comment:
Is this just to remove weak check churn?

Reply:
Yes, locking weak ptrs is actually very espensive. Since the RenderPass is immediately doing the recording and using the ContextVK to allocate descriptor sets, it would potentially lock multiple times per command otherwise.